File: 1291-sequential-digits/solution.cpp
#include <vector>
#include <string>
#include <algorithm>

class Solution {
public:
    std::vector<int> sequentialDigits(int low, int high) {
        std::vector<int> result;
        std::string digits = "123456789";
        
        // Iterate over all possible lengths of sequential numbers
        for (int length = 2; length <= 9; ++length) {
            // Iterate over all possible starting positions
            for (int i = 0; i <= 9 - length; ++i) {
                std::string sub = digits.substr(i, length);
                int num = std::stoi(sub);
                
                // If the generated number is within range, add to results
                if (num >= low && num <= high) {
                    result.push_back(num);
                }
            }
        }
        
        // Although generation order naturally produces sorted output, 
        // explicit sorting ensures correctness if generation order changes.
        std::sort(result.begin(), result.end());
        return result;
    }
};

File: 1291-sequential-digits/solution.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> list[int]:
        result = []
        digits = "123456789"
        
        # A sequential number cannot be longer than 9 digits (1-9)
        # We iterate through all possible lengths (2 to 9)
        for length in range(2, 10):
            # We iterate through all valid start indices for the given length
            for i in range(9 - length + 1):
                # Extract the sequential string
                num = int(digits[i : i + length])
                
                # Check bounds
                if low <= num <= high:
                    result.append(num)
                elif num > high:
                    # Since we iterate lengths and starts in order, 
                    # we can potentially break early if num exceeds high
                    break
        
        # Results are appended in sorted order
        return result